### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Paskab012**
Hello ✋🏽 , 
In this PR I have implemented the following : 
## GENERAL TASK COMPLETED
- Implement classes and objects in Ruby.
- Implement encapsulation and inheritance with Ruby.
- Run a program using the command line.
## DETAILED TASKS IMPLEMENTED
- I have created class `Person` with the following: Instance vars: `@id`, `@name`, and `@age`, a constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true. Getters for `@id`, `@name`, and `@age`. Setters for `@name` and `@age.`
Private method `is_of_age?` that returns true if `@age` is greater or equal to 18 and false otherwise.
Public method `can_use_services?` that returns true if `person` is of age or if they have permission from parents.

- I have created class Student with the following: Inherits from Person. Constructor extends parent's constructor by adding `@classroom ` and a parameter for it. Method `play_hooky` that returns "¯\(ツ)/¯".

- Create class Teacher with the following: Inherits from Person. Constructor extends parent's constructor by adding `@specialization ` and a parameter for it. Override `can_use_services?` so it always returns true.

- I made sure each class Is saved in a separate file.

### CODE REVIEW

**Review by Kingstalux — COMMENTED** [review]
Hi @Paskab012 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌ 

- [ ] You have submitted a project, that has failing linters checks. Kindly make sure your linters are passing before submitting your project for a review 👍 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Paskab012** [comment]
Hello @Kingstalux , 
This is actually what I am getting in the `Terminal` when I fix linters errors with `Rubocop`

Screen Shot 2022-05-17 at 13 04 07

File: per_son.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
```
Comment (Mihndim2020) [inline per_son.rb:10]: - [OPTIONAL] For better code organization, it will be nice to move this `setter and getter` methods to the top of your class.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
### Hi @Paskab012, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**